sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BeamRowToStorageApiProto.java

```diff
@@ -243,7 +245,19 @@ private static TableFieldSchema fieldDescriptorFromBeamField(Field field) {
         builder = builder.setType(type);
         break;
       case MAP:
-        throw new RuntimeException("Map types not supported by BigQuery.");
+        @Nullable FieldType keyType = field.getType().getMapKeyType();
+        @Nullable FieldType valueType = field.getType().getMapValueType();
+        if (keyType == null || valueType == null) {
+          throw new RuntimeException("Unexpected null element type!");
```

Comment:
Can you add some context to this exception around the error being in converting to the storage api proto? That would help users diagnose their pipelines without needing to know the beam code as well